lib/spack/spack/util/git.py

```diff
@@ -18,6 +18,9 @@
 # regex for a commit version
 COMMIT_VERSION = re.compile(r"^[a-f0-9]{40}$")
 
+# regex for (git) version
+GIT_VERSION = re.compile(r"(\d+(?:\.\d+)*(?:[-+a-zA-Z0-9.]*))")
```

Comment:
Do we care about this matching potentially ill formed version strings? If not, this is fine, otherwise, we should use semver's regex and take just the numeric component.

Reply:
We haven't yet so I'm more partial to a simplified regex.